src/common/gui/base/SnackBar.ts

```diff
@@ -48,8 +49,10 @@ function makeButtonAttrsForSnackBar(button: SnackBarButtonAttrs): ButtonAttrs {
  * @param message The message to be shown. It must be short enough to ensure it is always shown in 2 lines of text at max in any language.
  * @param snackBarButton will close the snackbar if it is clicked (onClose() will be called)
  * @param onClose called when the snackbar is closed (either by timeout or button click)
+ * @param waitingTime number of milliseconds to wait before showing the snackbar
  */
-export function showSnackBar(args: { message: TranslationText; button: SnackBarButtonAttrs; onClose?: () => void }) {
+export async function showSnackBar(args: { message: TranslationText; button: SnackBarButtonAttrs; onClose?: () => void; waitingTime?: number }) {
+	if (args.waitingTime) await delay(args.waitingTime)
```

Comment:
We have a ```debounce``` function inside utils. I'd say that it's better to debounce the showSnackBar call than waiting inside of it